content/hardware/05.pro-solutions/solutions-and-kits/portenta-machine-control/tutorials/pmc-opta-modbus-tcp/content.md

```diff
@@ -0,0 +1,288 @@
+
+#### Modbus TCP - Client
+
+With the Opta™ successfully connected to the PLC IDE, it is time to configure the Modbus TCP communication.
+
+In the **Ethernet** tab, enable the Modbus TCP Master mode.
+
+![Modbus TCP Master mode enabled](assets/master-modbus.png)
+
+Then right-click on the **Ethernet** tab, click on **Add** and select the _Generic Modbus device_.
+
+![Modbus Device Configuration 1](assets/modbus-device.png)
+
+On the `Generic Modbus_1` device settings, enter the Server IP, the one from the Portenta Machine Control.
+
+![Modbus Device Configuration 2](assets/modbus-device-2.png)
+
+Right-click on the device and add the **FC-04 Modbus** function that will let us read the server input registers. 
+
+![Modbus Device Configuration 3](assets/modbus-device-3.png)
+
+Now, click on the function and in the general settings, enter the shared variable address that you defined earlier in the server, `25000` in this case.
+
+![Modbus Device Configuration 4](assets/modbus-device-4.png)
+
+In the Input Register tab, create a label for it, which could be `temp_reg`.
+
+![Label the register](assets/modbus-device-5.png)
+
+Now, go to the **sketch editor** and uncomment the library and setup function code lines. As the IP, we must use the same as the Opta™. 
+
+In this case, the following configurations are used.
+
+```arduino
+// Enable usage of EtherClass, to set static IP address and other
+#include <PortentaEthernet.h>
+arduino::EthernetClass eth(&m_netInterface);
+
+void setup()
+{
+	// Configure static IP address
+	IPAddress ip(10, 0, 0, 227);    // Opta IP address
+	IPAddress dns(10, 0, 0, 1);     // gateway IP address
+	IPAddress gateway(10, 0, 0, 1); // gateway IP address
+	IPAddress subnet(255, 255, 255, 0); 
+	// If cable is not connected this will block the start of PLC with about 60s of timeout!
+	eth.begin(ip, dns, gateway, subnet);
+
+}
+```
+![Network settings for Modbus TCP](assets/ip-setup-opta.png)
+
+Finally, define the Opta™ outputs behavior in function of the temperature read from the Portenta Machine Control. For this, go to the **resources tab > Relay Outputs** and give a variable name to each relay, in this case, call them `relay_1`, 2, 3 and 4 respectively.
+
+![Outputs definitions](assets/outputs-opta.png)
+
+The same with the LED outputs, `LED1`, 2, 3 and 4.
+
+![LEDs definitions](assets/led-opta.png)
+
+Now, go to the main code in the Project tab. Create a variable by right-clicking on the local variables window and then **insert**, call it **temp_local** and _integer_ as the type.
+
+![Create a local variable](assets/variable-opta.png)
+
+In the code editor, match the local variable with the shared one sent by the Portenta Machine Control.
+
+Once stored locally, design the logic to control the outputs as you want. In this case, four different temperature levels will be set to control each output with the temperature rise respectively. Copy and paste the following script into the **main code** section.
+
+```
```

Comment:
FYI this is a technical limitation of the new docs, there is not syntax highlighting for this language.